app/code/Magento/Cms/Controller/Adminhtml/Page/PostDataProcessor.php

```diff
@@ -108,4 +113,26 @@ public function validateRequireEntry(array $data)
         }
         return $errorNo;
     }
+
+    /**
+     * Validate data, avoid cyclomatic complexity
+     *
+     * @param array $data
+     * @param \Magento\Framework\View\Model\Layout\Update\Validator $validatorCustomLayout
+     * @return bool
+     * @throws \Exception
+     */
+    private function validateData($data, $validatorCustomLayout)
+    {
+        $hasError = false;
+        if (!empty($data['layout_update_xml']) && !$validatorCustomLayout->isValid($data['layout_update_xml'])) {
+            $hasError = true;
+        }
+        if (!empty($data['custom_layout_update_xml']) &&
+            !$validatorCustomLayout->isValid($data['custom_layout_update_xml'])
+        ) {
+            $hasError = true;
+        }
+        return $hasError;
```

Comment:
I would consider a `return false;` and then we could remove the `$hasError` variable here.